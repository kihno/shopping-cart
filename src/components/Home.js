import React from 'react';
import ShopButton from './ShopButton';

const Home = (props) => {
    const { handleShopClick } = props;

    return(
        <div id="home">
            <h1>UNSC Surplus</h1>
            <h2>Travel the ring in style</h2>
            <ShopButton handleShopClick={handleShopClick} />
        </div>
    )
}

export default Home;