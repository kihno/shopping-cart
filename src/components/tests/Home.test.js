import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../Home';

describe('Home page', () => {
    it('renders home page', () => {
        const { container } = render(<Home />);
        expect(container).toMatchSnapshot();
    });

    it('renders shop button', () => {
        const {getByRole } = render(<Home />);

        expect(getByRole('button').textContent).toMatch(/shop now/i);
    });
});