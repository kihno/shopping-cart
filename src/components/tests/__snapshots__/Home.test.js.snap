// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders home page 1`] = `
<div>
  <div
    id="home"
  >
    <h1>
      Home Page
    </h1>
    <button
      id="shopButton"
      type="button"
    >
      Shop Now
    </button>
  </div>
</div>
`;
